### ISSUE TITLE
`tune.exfat` and `exfatlabel` don't print an error when setting an invalid volume serial

### ISSUE THREAD

**Issue by a1346054** [exfatprogs/exfatprogs#228]
Both `tune.exfat` and `exfatlabel` don't print an error when setting an invalid volume serial.

exfatprogs version : 1.2.1

Create a container:
```
fallocate -l 1G ./mycontainer
```

Format it as exFAT:
```
mkfs.exfat ./mycontainer
```

Display volume serial:
```
tune.exfat -i ./mycontainer | grep serial
```

should give something like `volume serial : 0x66fb7876`

Attempt to set an invalid serial (serial numbers are supposed to be hex digits):
```
tune.exfat -I 0x123qq678 ./mycontainer
```
The command completes successfully. `tune.exfat -i ./mycontainer | grep serial` reports `volume serial : 0x123` which shows that the serial was truncated. The command should instead report an error and not proceed with setting any serial number.

Additionally, if you try to set a too long serial number:
```
tune.exfat -I 0x1234567890abcd ./mycontainer
```

then `tune.exfat -i ./mycontainer | grep serial` will give `volume serial : 0x7890abcd` which is the last 8 digits of the serial that was attempted to be set. The command should instead report an error and not proceed with setting any serial number.

**Comment by namjaejeon** [comment]
@a1346054 Can you make the patch for this ?

**Comment by a1346054** [comment]
I'll have a look but I'm not much of a C coder.

The right place to fix it is probably inside lib/libexfat.c and not inside the individual tools.

The way I'm thinking of fixing it is doing a check inside exfat_set_volume_serial() on the format of ui->volume_serial and if the check fails then call exfat_err() with some suitable error message and ret = -1

**Comment by namjaejeon** [comment]
@a1346054 

> The right place to fix it is probably inside lib/libexfat.c and not inside the individual tools.

Right.

> The way I'm thinking of fixing it is doing a check inside exfat_set_volume_serial() on the format of ui->volume_serial and if the check fails then call exfat_err() with some suitable error message and ret = -1

Okay.

And Can you check how other filesystem utils handle it ? they make the error or truncate...

**Comment by namjaejeon** [comment]
@a1346054 Please close if it is improved on new exfatprogs-1.2.3 version.

**Comment by HanabishiRecca** [comment]
Umm, guys, it is now vice versa and doesn't accept **any** serials whatsoever. Even ones it literally printed itself.

```
# tune.exfat -i /dev/sdb1
exfatprogs version : 1.2.3
volume serial : 0xf7fef534
```
```
# tune.exfat -I 0xf7fef534 -v /dev/sdb1
invalid serial number(0xf7fef534)
```

**Comment by namjaejeon** [comment]
@HanabishiRecca 

Can you check your issue with #exfat-next ? this patch in #exfat-next seems to fix your issue. (https://github.com/exfatprogs/exfatprogs/commit/11dff3540156eb12075f97dfd31f8e26c91dd957)

git clone https://github.com/exfatprogs/exfatprogs --branch=exfat-next

**Comment by HanabishiRecca** [comment]
Yeah, it works.

**Comment by namjaejeon** [comment]
@hclee We need to release 1.2.4 version for this issue.

**Comment by hclee** [comment]
> @hclee We need to release 1.2.4 version for this issue.

I will do tomorrow because of security issue.

**Comment by hclee** [comment]
Hello @HanabishiRecca 
We have released 1.2.4 version, which includes fixing the issue.

**Comment by HanabishiRecca** [comment]
I got the update already in Arch, thank you.
